IOMC/EventVertexGenerators/src/HLLHCEvtVtxGenerator.cc

```diff
@@ -48,35 +47,77 @@ void HLLHCEvtVtxGenerator::fillDescriptions(edm::ConfigurationDescriptions& desc
+    fTimeOffset = p.getParameter<double>("TimeOffsetInns") * CLHEP::ns * c_light;
+    fEProton = p.getParameter<double>("EprotonInGeV") * 1e9;
+    fCrossingAngle = p.getParameter<double>("CrossingAngleInurad") * 1e-6;
+    fCrabFrequency = p.getParameter<double>("CrabFrequencyInMHz") * 1e6;
+    fRF800 = p.getParameter<bool>("RF800");
+    fBetaCrossingPlane = p.getParameter<double>("BetaCrossingPlaneInm");
+    fBetaSeparationPlane = p.getParameter<double>("BetaSeparationPlaneInm");
+    fHorizontalEmittance = p.getParameter<double>("HorizontalEmittance");
+    fVerticalEmittance = p.getParameter<double>("VerticalEmittance");
+    fBunchLength = p.getParameter<double>("BunchLengthInm");
+    fCrabbingAngleCrossing = p.getParameter<double>("CrabbingAngleCrossingInurad") * 1e-6;
+    fCrabbingAngleSeparation = p.getParameter<double>("CrabbingAngleSeparationInurad") * 1e-6;
+    // Set parameters inferred from configurables
+    gamma = fEProton / pmass + 1.0;
+    beta = std::sqrt((1.0 - 1.0 / gamma) * ((1.0 + 1.0 / gamma)));
+    betagamma = beta * gamma;
+    oncc = fCrabbingAngleCrossing / fCrossingAngle;
+    epsx = fHorizontalEmittance / (betagamma);
+    epss = epsx;
+    sigx = std::sqrt(epsx * fBetaCrossingPlane);
+    phiCR = oncc * fCrossingAngle;
+  }
+  if (readDB_) {
+    // NOTE: this is currently watching LS transitions, while it should watch Run transitions,
+    // even though in reality there is no Run Dependent MC (yet) in CMS
+    beamToken_ =
+        esConsumes<SimBeamSpotHLLHCObjects, SimBeamSpotHLLHCObjectsRcd, edm::Transition::BeginLuminosityBlock>();
+  }
+}
+
+void HLLHCEvtVtxGenerator::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const& iEventSetup) {
+  update(iEventSetup);
+}
+
+void HLLHCEvtVtxGenerator::update(const edm::EventSetup& iEventSetup) {
+  if (readDB_ && parameterWatcher_.check(iEventSetup)) {
+    edm::ESHandle<SimBeamSpotHLLHCObjects> beamhandle = iEventSetup.getHandle(beamToken_);
+    // Read configurable parameters
+    fMeanX = beamhandle->meanX() * CLHEP::cm;
+    fMeanY = beamhandle->meanY() * CLHEP::cm;
+    fMeanZ = beamhandle->meanZ() * CLHEP::cm;
+    fEProton = beamhandle->eProton() * 1e9;
+    fCrossingAngle = beamhandle->crossingAngle() * 1e-6;
+    fCrabFrequency = beamhandle->crabFrequency() * 1e6;
+    fRF800 = beamhandle->rf800();
+    fBetaCrossingPlane = beamhandle->betaCrossingPlane();
+    fBetaSeparationPlane = beamhandle->betaSeparationPlane();
+    fHorizontalEmittance = beamhandle->horizontalEmittance();
+    fVerticalEmittance = beamhandle->verticalEmittance();
+    fBunchLength = beamhandle->bunchLenght();
+    fCrabbingAngleCrossing = beamhandle->crabbingAngleCrossing() * 1e-6;
+    fCrabbingAngleSeparation = beamhandle->crabbingAngleSeparation() * 1e-6;
+    fTimeOffset = beamhandle->timeOffset() * CLHEP::ns * c_light;
+    // Set parameters inferred from configurables
+    gamma = fEProton / pmass + 1.0;
```

Comment:
@francescobrivio , if fEProton kinetic energy in GeV or total energy in GeV?

Reply:
The previous implementation was:
```
# In the header
const double momeV;

# In the constructor
momeV(p.getParameter<double>("EprotonInGeV") * 1e9)

# Used to initialize gamma
gamma(momeV / pmass + 1.0),
```

In the new implementation I've put:
```
# In the header
double fEProton; 

# In the constructor
fEProton = p.getParameter<double>("EprotonInGeV") * 1e9;

# Used to initialize gamma
gamma = fEProton / pmass + 1.0;
```

So technically everything is consistent, but indeed in the old implementation the `momeV` variable was _inferred_ from the energy by multiplying it by `1e9`, while now I've basically got rid of the `momeV` variable and I'm using directly the parameter read from the config.

Do you prefer me to fix this somehow?

Reply:
@francescobrivio , if EprotonInGeV is the total energy, then +1.0 is not needed. The difference is tiny but if you fix the class, then better to have a correct expression.

Reply:
What do you mean by "total energy" vs "kinetic energy" here?
`EprotonInGeV` is the beam energy, i.e. 6800 GeV or 7000 GeV I guess for Phase2.

Reply:
The difference is to use (gamma = fEProton / pmass + 1.0) or (gamma = fEProton / pmass) . Probably, nobody care about the difference - gamma = 7000 or gamma=7001 is not an important question. I suspect that (gamma = fEProton / pmass) is more correct and this should be defined somewhere in machine info.

Reply:
Ah ok! I'm not sure how to check that. Eventually in the future, when we get closer to HL-LHC, we'll have better information about the expected BeamSpot in general.

For the moment I would leave things unchanged, if you agree?